### DISCUSSION TITLE
new installation - configuration problems -

### DISCUSSION THREAD

**Discussion by pin32** [CrazyIvan359/mqttany#161] [Troubleshooting]
hi, I'm using an rpi4 for an openhab server and having problems with the gpio and mcp23017 bindings, looking online for a solution I came across this fantastic application.
I'll start by saying that I have no programming skills (beyond modifying some already structured code) so I probably left out something obvious (for those who know).
I tried to install mqttany on the rpi mentioned above, I followed the detailed guide from CrazyIvan (whom I thank for the time he dedicates to this project), but I encounter some malfunctions:

![immagine](https://github.com/CrazyIvan359/mqttany/assets/107401551/a09af64b-1294-4812-b4d9-add84a8c8903)

I guess I left something out in the configuration... but I need some help figuring out what. thanks to anyone who wants to give me a hand

**Comment by CrazyIvan359** [comment]
Hi @pin32 sorry for the late response, October has been wildly busy for me to date.

I believe you missed Step 3 from the [Installation](https://crazyivan359.github.io/mqttany/start.html#installation) instructions. Let me know if that doesn't sort things out or if you have any other issues! I'll try to be quicker getting back to you as well

**Comment by pin32** [comment]
![immagine](https://github.com/CrazyIvan359/mqttany/assets/107401551/aa5a9b14-4f50-4dbf-86f7-2da035fcc160)

I tried to redo the various steps but I still get the same error.
I don't know if it may depend on some other software that has access to gpios, I'm using a gpio binding for openhab...

**Reply by CrazyIvan359** [reply]
Are you running MQTTany as the root user? If so you would need to use `sudo` when running `pip` to install the requirements so they are installed for root.

As long as openHAB and MQTTany aren't trying to access the same pins there should be no issue running them both.

**Comment by pin32** [comment]
![immagine](https://github.com/CrazyIvan359/mqttany/assets/107401551/379b803c-73f7-45ae-907b-fcde038f9a46)

![immagine](https://github.com/CrazyIvan359/mqttany/assets/107401551/c8615aa2-5fb9-42fd-b55e-25f991b98027)

![immagine](https://github.com/CrazyIvan359/mqttany/assets/107401551/a9fbcbf1-7bc0-42fe-92a3-fcc29da06c15)

![immagine](https://github.com/CrazyIvan359/mqttany/assets/107401551/4e0deaa6-df7c-4926-8dd5-097927e394af)

it seems to be ok. I still see a strange message: (code=exited, status=1/FAILURE)
is it normal?

thanks for the support

**Reply by CrazyIvan359** [reply]
Ok, the documentation is a little unclear about it but you need to install requirements for all the modules you are using. That includes the namesake communication module 'MQTT'. The command for doing that is listed in the documentation for each module, so for the MQTT module it is `pip install -r requirements/mqtt.txt`.

---

![image](https://github.com/CrazyIvan359/mqttany/assets/6764025/adfd46d5-6f31-4d79-8b24-9aafb588ade5)
Here you are using `sudo` to run MQTTany, so this will definitely require using `sudo` when running `pip` also. What user did you put in the service file when you set that up, was it `root`?

---

Starting MQTTany using `sudo python3 mqttany/mqttany.py` will start a new copy of it that is *not* related to `systemctl` so the message you are seeing there is unrelated. It looks like you then pause and background the instance you just started using the above command though. 

Please go through the documentation for all the MQTTany modules you are using and make sure you install the required packages using `sudo pip install -r requirements/<NAME>.txt` and then reboot your Pi. After that run `sudo systemctl status mqttany` and see if it is now running.

In future, if you are working with a program that you check the status using `systemctl`, you can use `sudo systemctl restart <SERVICE>` to manually restart a service that you just changed some configuration or similar for, after that you can again request the status to see if it started correctly.
